enhancements/installer/ibm-cloud-for-power-vs-ipi.md

```diff
@@ -0,0 +1,381 @@
+against IBM Cloud APIs.
+- Implement the machine-api controllers to provide scaling and managing the cluster nodes post install.
+
+### Non-Goals
+
+- UPI support. We've already documented how to install on pp64le hardware using UPI.
+
+## Proposal
+
+In order to leverage the Power VS platform, we will make use of the Power VS support in the [IBM Cloud Terraform
+Provider][ibm-cloud-terraform-provider]. This provider will interface with IBM Cloud APIs and ultimately enable the
+OpenShift Installer to provision VM resources using Power VS, that will be used as worker and masters of the cluster.
+Unlike some of the existing providers, this depends on image availability of CoreOS in the Power VS cloud in the OVA
+format.
+
+The expected workflow would be as follows:
+1. The installer is invoked with credentials for ones IBM Cloud account.
+2. Once invoked, the installer interacts with a terraform provider to create a new Power VS service.
+3. The installer or terraform provider will then import a CoreOS OVA boot image file into the Power VS service from IBM
+Cloud Object Storage, which creates a corresponding boot image.
+4. Once imported the installer can start building the private network for the nodes.
+5. The bootstrap and master nodes will be created on in the private network. These are called Power VS instances, and
+they get created using the CoreOS boot image.
+6. In order to expose this private network and provide load balancing, the IBM Cloud APIs will leverage the IBM Cloud
+[Direct Link][ibm-direct-link-website] offering. This will allow for both the ingress load balancer and the cluster nodes to
+talk over the private network on a high-speed link.
+7. Bootstrapping completes as normal.
+8. The workers will be added using the openshift/[cluster-\*machine-\*] controllers and operators based on the openshift
+MAPI model. A Power VS controller needs to be implemented and hooked into the [OpenShift Machine
+Operator][openshift-machine-operator].
+9. The cluster teardown can make use of the existing [go client for Power][power-go-client] via `openshift-install
+destroy cluster`.
+
+### User Stories
+
+- A user of this installer would like to see a cluster running within an hour of kicking off the installer.
+- The expectation is that this is fully automated.
+
+### Implementation Details/Notes/Constraints
+
+#### Existing APIs in the Upstream
+There is a working [implementation of a machine API](https://github.com/openshift-powervs/cluster-api-provider-powervs)
+we've created for testing purposes. Though not ready for upstream, many have been tested and found to be working, such
+as mapi and nodeupdate controller. This was tested with machineset as well, so things look promising.
+
+#### Cloud Controller Manager
+We are planning to reuse the [cloud-provider-ibm][cloud-provider-ibm] for everything except for the node controller in
+machine api because the compute target doesn't support Power VS instances.
```

Comment:
@Karthik-K-N 

Is there a specific PR we should point to to make it more clear that the IBM Cloud CCMO supports PowerVS? This commit may be enough? https://github.com/openshift/cluster-cloud-controller-manager-operator/commit/d64dbaa7fc75685ca46d23110fd5b57ca9f4aa38

And would it be accurate to add, "The node objects are still managed by [cluster-api-provider-powervs](https://github.com/openshift/cluster-api-provider-powervs)".

Reply:
this is the PR that adds powervs support to CCM where it instantiates ibm cloud provider assets: https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/129

Reply:
I have submitted following PRs to use IBM Cloud CCMO and support Power VS
1. https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/129
2. https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/142
3. https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/148

and yes currently the node objects are still managed by cluster-api-provider-powervs:  [Node Controller](https://github.com/openshift/cluster-api-provider-powervs/blob/main/pkg/controller/nodeupdate/nodeupdate_controller.go)